### PR TITLE
Add github actions to run tests

### PR DESCRIPTION
**Pull request by kswanson33**

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -22,6 +22,7 @@ RUN apt-get update && \
     libicu-dev \
     && \
     gem install bundler --no-document && \
+    gem install rspec && \
```
Comment (jpslav) [inline docker/Dockerfile:25]: This is also down below on line 66 -- is it needed here?

File: docker/ci [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# critical to set CI=true because codecov-ruby won't detect GitHub Actions
+# without it per:
+# https://github.com/codecov/codecov-ruby/blob/484767f1c3d7992a9d7fedd6dc72d35a80d04f70/lib/codecov.rb#L68-L69
+
+docker run $CI_ENV \
+           -e CI=true \
+           -v $DIR/..:/code \
+           -w /code \
+           openstax/recipes.ci:latest \
+           /bin/bash -c "exec rspec"
```
Comment (jpslav) [inline docker/ci:15]: what is `exec` here?  Might have expected `bundle exec rspec` or just `rspec`

File: scripts/install_used_gem_versions [hunk 3]
```diff
@@ -4,14 +4,16 @@
...
+
 source_filenames = (Dir.glob(File.join(__dir__,"../books/**/*")) + [File.join(__dir__,"console"),File.join(__dir__,"normalize")])
                        .select{|filename| !File.directory?(filename)}
 
 gem_name_to_versions_array_hash = source_filenames.each_with_object({}) do |filename,hash|
   source = File.read(filename)
   gem_names_and_versions = source.scan(/gem\s+['"](\w+)['"]\s*,\s*['"](.+)['"]/)
   gem_names_and_versions.each do |gem_name, version|
-    (hash[gem_name] ||= []).push(version)
+    (hash[gem_name] ||= Set.new([])).add(version)
```
Comment (jpslav) [inline scripts/install_used_gem_versions:16]: good catch

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+    - cron: '0 0 * * 0' # weekly
```
Comment (jpslav) [inline .github/workflows/tests.yml:9]: is this needed? (does this mean it'll retest weekly?)  I have seen this in other folks' work too and don't see why it is here.

**Review by jpslav — CHANGES_REQUESTED** [review]

File: docker/ci (hunk 2 above)
Comment (kswanson33) [reply docker/ci:15]: not sure haha, I'll take it out

File: .github/workflows/tests.yml (hunk 4 above)
Comment (kswanson33) [reply .github/workflows/tests.yml:9]: this was copied directly from kitchen; I'll take it out

**Review by jpslav — APPROVED** [review]

**Pull request merged**